src-docs/src/views/overlay_mask/overlay_mask_example.js

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+
+import { renderToHtml } from '../../services';
+
+import { GuideSectionTypes } from '../../components';
+
+import { EuiOverlayMask, EuiCode } from '../../../../src/components';
+
+import OverlayMask from './overlay_mask';
+const overlayMaskSource = require('!!raw-loader!./overlay_mask');
+const overlayMaskHtml = renderToHtml(OverlayMask);
+
+const overlayMaskSnippet = `<EuiOverlayMask>
+  <!-- Content goes here -->
+</EuiOverlayMask>`;
+
+export const OverlayMaskExample = {
+  title: 'Overlay Mask',
+  sections: [
+    {
+      source: [
+        {
+          type: GuideSectionTypes.JS,
+          code: overlayMaskSource,
+        },
+        {
+          type: GuideSectionTypes.HTML,
+          code: overlayMaskHtml,
+        },
+      ],
+      text: (
+        <div>
+          <p>
+            <EuiCode>EuiOverlayMask</EuiCode> is an useful component to create
+            an overlay effect. It can be used to pop up messages. Best use of an
+            overlay mask is to wrap it on <EuiCode>EuiModal</EuiCode>. See
+            examples <a href="http://localhost:8030/#/layout/modal">Here</a>.
+          </p>
+          <p>
+            <EuiCode>EuiOverlayMask</EuiCode> accepts <EuiCode>onClick</EuiCode>
+            as prop, where function can be passed. It also accepts
+            <EuiCode>style</EuiCode> prop where string for custom styling is
+            passed.
```

Comment:
```suggestion
            There are <a href="https://www.nngroup.com/articles/overuse-of-overlays/">many considerations</a> to
            make before choosing to use an overlay. A the very least, 
            you must provide a visible button to close the overlay. You can also pass
            an <EuiCode>onClick</EuiCode> handler to handle closing the overlay 
            when they've clicked outside its content.
```

Reply:
"pass an onClick handler", do you mean by passing onClick to overlay mask? 
If yes, then "when they've clicked outside its content" contraditcts.

Even if we click inside content, overlay closes.

![closing_overlay](https://user-images.githubusercontent.com/17699717/77192199-31554580-6b02-11ea-8c64-e0aec68f45e6.gif)

Reply:
Hmm, yeah this is why documentation examples are important 😜 . I would not have expected the component to behave that way. I think the way we use it with flyouts is that the flyout is not a **child** of the overlay mask, but is a separate element...

```suggestion
            There are <a href="https://www.nngroup.com/articles/overuse-of-overlays/">many considerations</a> to
            make before choosing to use an overlay. A the very least, 
            you must provide a visible button to close the overlay. You can also pass
            an <EuiCode>onClick</EuiCode> handler to handle closing the overlay.
            However, be wary of using <EuiCode>onClick</EuiCode> and <EuiCode>children</EuiCode>, as clicking the <EuiCode>children</EuiCode>
            will also trigger the <EuiCode>onClick</EuiCode>.
```

Reply:
Fair enough.

Are they acceptable examples?

![closing_overlay gif](https://user-images.githubusercontent.com/17699717/77220019-9d1acb00-6b61-11ea-899b-057756304087.gif)

Reply:
Those look great, can you push your changes?